adsingestp/parsers/jats.py

```diff
@@ -190,24 +191,58 @@ def parse(self, article_metadata):
         # JATS puts author data in <contrib-group>, giving individual authors in each <contrib>
         for art_group in art_contrib_groups:
             art_contrib_group = art_group.extract()
-            contribs_raw = art_contrib_group.find_all("contrib")
+
+            # extract <contrib> from each <contrib-group>
+            contribs_raw_init = []
+            if art_contrib_group.find_all("contrib", recursive=False):
+                contribs_raw_init = art_contrib_group.find_all("contrib", recursive=False)
+
             default_key = "ALLAUTH"
+
+            # cycle through <contrib> to check if a <collab> is listed in the same level as an author an has multiple authors nested under it; targeted for Springer
+            contribs_raw = []
+            collabflag = None
+            for i in range(len(contribs_raw_init)):
```

Comment:
You should be able to structure this as: `for contrib in contribs_raw_init`, which keeps you away from having to using indices everywhere. You do save off the index in `collabflag`, so you do need to access the index, but it's generally more common to write it this way: `for idx, contrib in enumerate(contribs_raw_init)` - then you use `contrib` to access the item itself, and only use `idx` when you specifically need to access the index.

Reply:
I took your advice and attempted a modification. It even caught some collabs that I missed the first time around. It works well now.